backend/gce.go

```diff
@@ -644,10 +644,15 @@ func (p *gceProvider) apiRateLimit(ctx gocontext.Context) error {
 
 		// Sleep for up to 1 second
 		var span *trace.Span
-		ctx, span = trace.StartSpan(ctx, "GCE.timeSleep.apiRateLimit")
+		if trace.FromContext(ctx) != nil {
+			ctx, span = trace.StartSpan(ctx, "GCE.timeSleep.apiRateLimit")
+		}
+
 		time.Sleep(time.Millisecond * time.Duration(mathrand.Intn(1000)))
```

Comment:
I kept this `time.Sleep` outside of any conditionals, because we still need it to run independent of whether a context is present or not for tracing.